pkg/controller/v1beta1/inferenceservice/reconcilers/keda/keda_reconciler.go

```diff
@@ -156,8 +156,12 @@ func getKedaMetrics(componentExt *v1beta1.ComponentExtensionSpec, configMap *cor
 
 				if triggerType == string(constants.AutoScalerMetricsSourceOpenTelemetry) {
 					trigger.Type = "external"
+					// Inject namespace and deployment label selectors into the query for metric isolation.
+					// This ensures the metricQuery only selects metrics for the correct deployment and namespace.
+					// Example: sum(<query>{namespace="<namespace>", deployment="<deployment>"})
+					metricQuery := fmt.Sprintf("sum(%s{namespace=\"%s\", deployment=\"%s\"})", query, componentMeta.Namespace, componentMeta.Name)
```

Comment:
The query itself can include the sum and labels already? not sure if we can inject this way and still ensure a valid query.

Reply:
OK, it is possible that users specify only metrics with this parameter, or specify a full query with this parameter. How can we tell the differences?

Reply:
on Autoscaling spec there are both `metricNames` and `query` field
https://github.com/kserve/kserve/blob/master/pkg/apis/serving/v1beta1/component.go#L305. I think if the user only specifies the metricsNames then we can default the query with your logic. If query field is specified we do not change it.

Reply:
It seems this `query` was taken as the metric in either the test and the example.
I will keep the `query` the same as before, but for metricNames, I will add the labels of deployment name and namespace.

Reply:
will create a separate PR to fix this